src/main.js

```diff
@@ -1,22 +1,264 @@
+    try{
+        Cypress.env('totalPokemon', total);
 
+    } catch(error){
+        console.log(error);
+    }
+}
+function getPokemonHandler(pokemon){
+    clearPokemon();
+    addLoadingToPokemon();
+    hidePokemonNotFoundError()
+    getPokemonByIdOrName(pokemon);
+}
+function addLoadingToPokemon(){
+    $('#pokemon-carousel .carousel-inner')
+    .append(`<div class="loader pokemon-carousel active"></div>`)
+}
+function removeLoadingFromPokemon(){
+    $('#pokemon-carousel .loader.pokemon-carousel').remove();
+}
+let totalPokemon // total amount of pokemons is global por testing purposes
+let firstPokemonOnListId;
+let lastPokemonOnListId;
+function updateFirstAndLastPokemon(offset, amount){
+    firstPokemonOnListId = offset + 1;//first offset is 0 but first pokemon is 1
+    lastPokemonOnListId = offset + amount;
+    if(lastPokemonOnListId > totalPokemon)//makes sure it doesen't go further from last pokemon
+    lastPokemonOnListId = totalPokemon;
+}
+function updatePokemonList(pokemonList, offset){
+    clearPokemonList();
+    let i = 0;
+    for(const pokemon of pokemonList){
+        i++
+        addPokemonToList(pokemon, offset+i);
+    }
+    updatePageIndicator();
+}
+function updatePageIndicator(){
+    const pageIndicator = $('#pokemon-list-page');
+    pageIndicator.text(`pokemon ${firstPokemonOnListId} to ${lastPokemonOnListId}`);
+}
+function addPokemonToList(pokemon, index){
+    const pokemonButton = $(`<button class="btn btn-link" id="${pokemon.name}">${index}. ${pokemon.name}</button>`);
+    const li = $('<li></li>')
+    li.append(pokemonButton);
+    $('#pokemon-list').append(li);
+    addLoadingToListItem(li)
+    fetchSpriteToList(li, pokemon);
+}
+function addLoadingToListItem(element){
+    const loading = $(`<div class="lds-dual-ring"></div>`);
+    loading.addClass('lds-dual-ring');
+    element.append(loading);
+}
+function removeLoadingFromListItem(element){
+    element.children('.lds-dual-ring').remove();
+}
+function fetchSpriteToList(element, pokemon){
+    fetch(`https://pokeapi.co/api/v2/pokemon/${pokemon.name}`)
+    .then(response => response.json())
+    .then(data => {
+        addSprite(data.sprites.front_default, element)
+        removeLoadingFromListItem(element)
+    })
+    .catch(() => {
+        alert('Could not get pokemon for sprite list');
+    })
+}
+function addSprite(sprite, htmlItem){
+    if(sprite === null) return;
+    const img = $(`<img src="${sprite}" class="list-sprite">`);
+    htmlItem.append(img);
+}
+function clearPokemonList(){
+    const list = $('#pokemon-list');
+    list.children().remove();
+}
 
+function getPokemonByIdOrName(pokemon){
+    if(pokemon === 'michelin'){
+        easterEgg();
+        return;
+    }
+    fetch(`https://pokeapi.co/api/v2/pokemon/${pokemon}`)
+    .then(response => response.json())
+    .then(data => {
+        pokemonHandler(data);
+    })
+    .catch(() => {
+        showPokemonNotFoundError(pokemon);
+        clearPokemon();
+    });
+}
```

Comment:
I think it only should get a Pokemon (fetch it), also it should receive a pokemonId or a pokemonName, receveing a pokemon makes me think that it receives a pokemon entity.